packages/ketcher-core/src/application/render/restruct/resimpleObject.ts

```diff
@@ -197,56 +207,61 @@ class ReSimpleObject extends ReObject {
     });
     const scaleFactor = render.options.scale;
 
-    const path: Array<any> = [];
+    const paths: Array<StyledPath> = [];
 
     // TODO: It seems that inheritance will be the better approach here
     switch (this.item.mode) {
       case SimpleObjectMode.ellipse: {
         const rad = Vec2.diff(point[1], point[0]);
         const rx = rad.x / 2;
         const ry = rad.y / 2;
-        path.push(
-          render.paper.ellipse(
-            tfx(point[0].x + rx),
-            tfx(point[0].y + ry),
-            tfx(Math.abs(rx) + scaleFactor / 8),
-            tfx(Math.abs(ry) + scaleFactor / 8),
-          ),
+        const outerEllipse = render.paper.ellipse(
+          tfx(point[0].x + rx),
+          tfx(point[0].y + ry),
+          tfx(Math.abs(rx) + scaleFactor / 8),
+          tfx(Math.abs(ry) + scaleFactor / 8),
         );
+        paths.push({
+          path: this.getHoverPathStyle(outerEllipse, render, true),
+          stylesApplied: true,
+        });
         if (
           Math.abs(rx) - scaleFactor / 8 > 0 &&
           Math.abs(ry) - scaleFactor / 8 > 0
         ) {
-          path.push(
-            render.paper.ellipse(
-              tfx(point[0].x + rx),
-              tfx(point[0].y + ry),
-              tfx(Math.abs(rx) - scaleFactor / 8),
-              tfx(Math.abs(ry) - scaleFactor / 8),
-            ),
+          const innerEllipse = render.paper.ellipse(
+            tfx(point[0].x + rx),
+            tfx(point[0].y + ry),
+            tfx(Math.abs(rx) - scaleFactor / 8),
+            tfx(Math.abs(ry) - scaleFactor / 8),
           );
+          paths.push({
+            path: this.getHoverPathStyle(innerEllipse, render, false),
+            stylesApplied: true,
+          });
         }
         break;
       }
 
       case SimpleObjectMode.rectangle: {
-        path.push(
-          render.paper.rect(
-            tfx(Math.min(point[0].x, point[1].x) - scaleFactor / 8),
-            tfx(Math.min(point[0].y, point[1].y) - scaleFactor / 8),
-            tfx(
-              Math.max(point[0].x, point[1].x) -
-                Math.min(point[0].x, point[1].x) +
-                scaleFactor / 4,
-            ),
-            tfx(
-              Math.max(point[0].y, point[1].y) -
-                Math.min(point[0].y, point[1].y) +
-                scaleFactor / 4,
-            ),
+        const outerRect = render.paper.rect(
+          tfx(Math.min(point[0].x, point[1].x) - scaleFactor / 8),
+          tfx(Math.min(point[0].y, point[1].y) - scaleFactor / 8),
+          tfx(
+            Math.max(point[0].x, point[1].x) -
+              Math.min(point[0].x, point[1].x) +
+              scaleFactor / 4,
+          ),
+          tfx(
+            Math.max(point[0].y, point[1].y) -
+              Math.min(point[0].y, point[1].y) +
+              scaleFactor / 4,
```

Comment:
If you understood what "tfx" is please rename it

Reply:
I found there is a separate Jira for this https://github.com/epam/ketcher/issues/2675, so I just skip it for now

Reply:
I found there is a separate Jira for this https://github.com/epam/ketcher/issues/2675, so I just skip it for now

Reply:
I found there is a separate Jira for this https://github.com/epam/ketcher/issues/2675, so I just skip it for now